mlrun/common/schemas/artifact.py

```diff
@@ -51,10 +53,44 @@ class ArtifactIdentifier(pydantic.BaseModel):
     key: typing.Optional[str]
     iter: typing.Optional[int]
     uid: typing.Optional[str]
+    tree: typing.Optional[str]
     # TODO support hash once saved as a column in the artifacts table
     # hash: typing.Optional[str]
 
 
 class ArtifactsFormat(mlrun.common.types.StrEnum):
     full = "full"
     legacy = "legacy"
+
+
+class ArtifactMetadata(pydantic.BaseModel):
+    key: str
+    project: str
+    iter: typing.Optional[int]
+    tree: typing.Optional[str]
+    tag: typing.Optional[str]
+
+    class Config:
+        extra = pydantic.Extra.allow
+
+
+class ArtifactSpec(pydantic.BaseModel):
+    src_path: typing.Optional[str]
+    target_path: typing.Optional[str]
+    viewer: typing.Optional[str]
+    inline: typing.Optional[str]
+    format: typing.Optional[ArtifactsFormat]
```

Comment:
is legacy format still a thing?

Reply:
Not really, but old clients might still use this param